examples/custom_bazel_integration_test/custom_bazel_integration_test.bzl

```diff
@@ -0,0 +1,39 @@
+load(
+    "@contrib_rules_bazel_integration_test//bazel_integration_test:defs.bzl",
+    "bazel_integration_test",
+)
+
+load("@cgrindel_bazel_starlib//shlib/rules:execute_binary.bzl", "execute_binary")
+
+load(
+    "//:bazel_versions.bzl",
+    "CURRENT_BAZEL_VERSION",
+)
+
+def custom_bazel_integration_test(
+        name,
+        test_runner,
+        workspace_path,
+        **kwargs):
+
+    native.sh_binary(
+        name = "%s_wrapper" % name,
+        srcs = ["//:test_runner_wrapper.sh"],
+        deps = [
+            "@bazel_tools//tools/bash/runfiles",
+        ],
+        data = [
+            "@contrib_rules_bazel_integration_test//tools:create_scratch_dir",
+        ],
+        testonly = True,
+    )
+
+    bazel_integration_test(
+        name = name,
+        bazel_version = CURRENT_BAZEL_VERSION,
+        test_runner = "%s_wrapper" % name,
+        test_runner_args = ["--test-runner", "$(location %s)" % test_runner],
+        workspace_path = workspace_path,
+        data = [test_runner],
```

Comment:
Continued from the other comment thread. 

I see the `test_runner_args` and `data` in the example. What I was inarticulately trying to say was that I don't see how this example demonstrates new functionality for the ruleset. What would be an example of an argument that you want to pass to `bazel_integration_test` that cannot happen with the current version of the rules?

Reply:
> What would be an example of an argument that you want to pass to bazel_integration_test that cannot happen with the current version of the rules?

This whole example would not be possible with current version of rules. I just thought create customization of test rule with common setup procedure, but still providing opportunity to write runner in any language, would be elegant demonstration how this attributes could be useful. Maybe I should come up with another example which will outline it better?

Reply:
I think so. Here is how I understand the current state:

- In the main branch, we have examples of writing the runner in different languages ([Python](https://github.com/bazel-contrib/rules_bazel_integration_test/blob/main/bazel_integration_test/py/bazel_py_integration_tests.bzl#L9) and [Swift](https://github.com/bazel-contrib/rules_bazel_integration_test/tree/main/examples/custom_test_runner)). These runners work without needing to plumb through command-line arguments passed to the `bazel_integration_test`.
- As you pointed out, there is a facility for passing different arguments to the test runner using `execute_binary`. Is the make variable substitution the only limitation? If so, perhaps we could add support for that.
- My original thinking when putting this ruleset together was that the runner does the heavy lifting (e.g. process args, set up the test env, execute the assertions). The `bazel_integration_test` provides the Bazel binary, sets up a test sandbox, and handles a failure of the runner as a test failure.
- In terms of how someone would use this new functionality, I would have expected to see arguments being specified [here](https://github.com/bazel-contrib/rules_bazel_integration_test/pull/65/files#diff-57350d429589f2e0494afc02e082a61fd3099a5654eb1b6b56139c39869c75f2R10) perhaps using make-variable substitution. As it stands, I don't see an example that could not have been implemented without this PR.
- In retrospect, I should have asked you to show an example of what you wanted to achieve.